### PR TITLE
feat: Rework user resource

### PR DESCRIPTION
**Pull request by sfc-gh-asawicki**
Rework user resource:
- schema adjusted
- handle show output properly
- updated user acceptance tests with new builders and assertions
- introduce new helpers in `resource_helpers_create.go`, `resource_helpers_update.go`,  and `resource_helpers_read.go`
- cleaned parameters handling a bit
- reused `parameterDef` between user and schema
- removed old way of handling keys in tests
- added float property to the SDK
- added integration tests documenting specific attributes behavior

Next PRs:
- handle policies for user (SNOW-1645875)
- deal with improper custom diffs with diff suppression (SNOW-1629468)
- move collection util (SNOW-1473414)
- remove `checkBool` helper
- datasource
- rename methods in `resource_helpers_read.go`
- update all changes in README (+ note that external changes for a few attributes)
- change description of user public keys resource (should be used only if user is not managed by terraform)
- add one more test for login and display name
- check the exact behavior of default_namespace and default_role because it looks like it is handled in a case-insensitive manner on Snowflake side
- check if any user parameter needs diffsuppression or validation
- test if hasMfa is always non-nullable, check if this has mfa helps with disable mfa, add to the describe output too
- test what is visible if everything is set if there is no ownership on the active role (+ update docs)
- handle `snowflake_user_password_policy_attachment` and `snowflake_user_public_keys`
- add `IgnoreChangeToCurrentSnowflakeValueInShow` and other suppressors

References:
- #1155 
- #1572

### CODE REVIEW

File: pkg/resources/resource_helpers_create.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		parsed, err := booleanStringToBool(v)
+		if err != nil {
+			return err
+		}
+		*createField = sdk.Bool(parsed)
+	}
+	return nil
+}
+
+// TODO: NewAccountObjectIdentifierFromFullyQualifiedName versus one of the new functions?
```
Comment (sfc-gh-asawicki) [inline pkg/resources/resource_helpers_create.go:41]: @sfc-gh-jcieslak pls dm me when you get there

File: pkg/resources/user.go [hunk 2]
```diff
@@ -5,74 +5,138 @@ import (
...
-		Optional:    true,
-		Description: "Specifies the set of secondary roles that are active for the user’s session upon login. Currently only [\"ALL\"] value is supported - more information can be found in [doc](https://docs.snowflake.com/en/sql-reference/sql/create-user#optional-object-properties-objectproperties)",
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			ValidateDiagFunc: isValidSecondaryRole(),
+		},
+		DiffSuppressFunc: SuppressCaseInSet("default_secondary_roles"),
+		MaxItems:         1,
+		MinItems:         1,
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/user.go:129]: It works ;) check https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3026/files#diff-c56411787ea970a3740f6f0252b3ec7ec26c6a173e98efad897190fdac606128R1082

File: pkg/resources/resource_helpers_create.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+func accountObjectIdentifierAttributeCreate(d *schema.ResourceData, key string, createField **sdk.AccountObjectIdentifier) error {
+	if v, ok := d.GetOk(key); ok {
+		*createField = sdk.Pointer(sdk.NewAccountObjectIdentifierFromFullyQualifiedName(v.(string)))
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/resource_helpers_create.go:44]: Thx!

File: pkg/resources/resource_helpers_create.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	return nil
+}
+
+// TODO: DecodeSnowflakeParameterID versus one of the new functions?
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/resource_helpers_create.go:49]: Hmmm, okay, I think we can stick with any of these two for the time-being and observe. Would it be okay?

File: pkg/resources/user.go [hunk 5]
```diff
@@ -5,74 +5,138 @@ import (
...
+		Description:      booleanStringFieldDescription("Specifies whether the user is forced to change their password on next login (including their first/initial login) into the system."),
+		Default:          BooleanDefault,
 	},
 	"disabled": {
-		Type:     schema.TypeBool,
-		Optional: true,
-		Computed: true,
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validateBooleanString,
```
[the remark this answers is not in the document]
Comment (sfc-gh-asawicki) [reply pkg/resources/user.go:85]: It should be used for more attributes. I will add this as another TODO for the next PR, oky?

File: pkg/resources/user.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/resources/user.go:130]: Hmm it doesn't move me to the particular file but whole diff, I was just referring to the documentation on MinItems that is saying 
> // If the field Optional is set to true then MinItems is ignored and thus
> // effectively zero.
If that's not true then we can leave it that way.

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**